src/pages/groupComparison/GroupComparisonStore.ts

```diff
@@ -0,0 +1,84 @@
+import {ResultsViewPageStore} from "../resultsView/ResultsViewPageStore";
+import {remoteData} from "../../shared/api/remoteData";
+import ListIndexedMap from "../../shared/lib/ListIndexedMap";
+import {Sample, SampleIdentifier} from "../../shared/api/generated/CBioPortalAPI";
+import { computed } from "mobx";
+import { PatientSurvival } from "shared/model/PatientSurvival";
+import * as _ from "lodash";
+import { COLORS } from "pages/studyView/StudyViewUtils";
+
+export default class GroupComparisonStore {
+    constructor(
+        private resultsViewPageStore:ResultsViewPageStore
+    ) {
+    }
+
+    public readonly sampleGroups = remoteData({
+        await:()=>[
+            this.resultsViewPageStore.samples,
+            this.resultsViewPageStore.sampleGroups
+        ],
+        invoke:()=>{
+            const sampleIdentifierToSample = new ListIndexedMap<Sample>();
+            for (const sample of this.resultsViewPageStore.samples.result!) {
+                sampleIdentifierToSample.set(sample, sample.sampleId, sample.studyId);
+            }
+
+            let colorIndex = 0;
+            return Promise.resolve(
+                this.resultsViewPageStore.sampleGroups.result!.map((group)=>({
+                    name: group.name,
+                    samples: _.reduce(group.sampleIdentifiers,(samples:Sample[], si:SampleIdentifier)=>{
+                        if(sampleIdentifierToSample.get(si.sampleId, si.studyId)){
+                            samples.push(sampleIdentifierToSample.get(si.sampleId, si.studyId)!);
+                        }
+                        return samples;
+                    },[]),
+                    color: group.color ? group.color : COLORS[(colorIndex++) % COLORS.length],
+                    value: group.name,
+                    legendText: group.legendText
+                }))
+            );
+        }
+    });
+
+    public readonly patientToAnalysisGroups = remoteData({
+        await: () => [
+            this.sampleGroups
+        ],
+        invoke: () => {
+            return Promise.resolve(_.reduce(this.sampleGroups.result, (acc, next) => {
+                next.samples.forEach(sample => {
+                    let groups = acc[sample.uniquePatientKey] || [];
+                    groups.push(next.name);
+                    acc[sample.uniquePatientKey] = groups;
+                })
+                return acc;
+            }, {} as { [id: string]: string []}));
+        }
+    });
+
+    @computed get showSurvivalTab() {
+        return this.resultsViewPageStore.survivalClinicalDataExists.isComplete && this.resultsViewPageStore.survivalClinicalDataExists.result;
```

Comment:
the store should get it's own survival data, not rely on resultsViewPageStore to get it